### PR TITLE
Fix cs storage tests

### PR DESCRIPTION
**Pull request by asubramanya**

### CODE REVIEW

File: src/backend/commands/tablecmds.c [hunk 1]
```diff
@@ -586,7 +586,8 @@ DefineRelation(CreateStmt *stmt, char relkind, char relstorage, bool dispatch)
 	}
 	else
 	{
-		stmt->ownerid = GetUserId();
+		if (!OidIsValid(stmt->ownerid))
```
Comment (asimrp) [inline src/backend/commands/tablecmds.c]: `DefineRelation()` is a general purpose function used for creating all kinds of relations.  Can you please add a comment explaining why this case is applicable only to split partition case?

**Comment by asimrp** [comment]
Can you please make commit message for "Fix the storage tests in tinc" more meaningful?  It currently lists tests that were fixed, but I can see a couple of tests being removed.

**Comment by ashwinstar** [comment]
Please can you file separate PRs focused on independent product fixes. And one generic for wholesale storage test fixes.

**Comment by asubramanya** [comment]
@ashwinstar I have separated the commits into separate PRs: #1494 and #1495

**Comment by asubramanya** [comment]
@asimrp Updated the commit message.

File: src/test/tinc/tincrepo/mpp/gpdb/tests/storage/vacuum/xidlimits/Makefile [hunk 2]
```diff
@@ -1,5 +1,10 @@
 MODULES=xidhelper
 PG_CONFIG=pg_config
 
+# Use what ever compiler is available on the system instead of 
+# the one reported by pg_config since it may not be available.
+override CC:=gcc
+override CFLAGS:=-m64 -O3 -fargument-noalias-global -fno-omit-frame-pointer -g -std=gnu99 -Wall -Wmissing-prototypes -Wpointer-arith -Winline -Wendif-labels -Wformat-security -fno-strict-aliasing -fwrapv -Wno-unused-but-set-variable -Wno-address
```
Comment (asimrp) [inline src/test/tinc/tincrepo/mpp/gpdb/tests/storage/vacuum/xidlimits/Makefile]: Are so many flags really needed?

Comment (asubramanya) [inline src/test/tinc/tincrepo/mpp/gpdb/tests/storage/vacuum/xidlimits/Makefile]: I have updated the PR to only use flags which are necessary.

**Comment by asubramanya** [comment]
Thanks for review. Pushed: 03126f4d3453976816e447f48db638121967ac62

**Pull request closed without merging**